finality-aleph/src/validator_network/manager/mod.rs

```diff
@@ -284,66 +272,67 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn outgoing() {
-        let mut manager = Manager::<Address, Data>::new();
+    async fn send_receive() {
+        let (mut connecting_id, _) = key().await;
+        let mut connecting_manager = Manager::<Address, Data>::new(connecting_id.clone());
+        let (mut listening_id, _) = key().await;
+        let mut listening_manager = Manager::<Address, Data>::new(listening_id.clone());
         let data = String::from("DATA");
-        let (peer_id, _) = key().await;
         let addresses = vec![
             String::from(""),
             String::from("a/b/c"),
             String::from("43.43.43.43:43000"),
         ];
         let (tx, _rx) = mpsc::unbounded();
         // try add unknown peer
-        manager.add_outgoing(peer_id.clone(), tx);
+        assert_eq!(
+            connecting_manager.add_connection(listening_id.clone(), tx),
+            Uninterested
+        );
         // sending should fail
         assert_eq!(
-            manager.send_to(&peer_id, data.clone()),
+            connecting_manager.send_to(&listening_id, data.clone()),
             Err(SendError::PeerNotFound)
         );
         // add peer, this time for real
-        assert!(manager.add_peer(peer_id.clone(), addresses.clone()));
+        if connecting_manager.add_peer(listening_id.clone(), addresses.clone()) {
+            assert!(!listening_manager.add_peer(connecting_id.clone(), addresses.clone()))
+        } else {
+            // We need to switch the names around, because the connection was randomly the
+            // other way around.
+            let temp_id = connecting_id;
+            connecting_id = listening_id;
+            listening_id = temp_id;
+            let temp_manager = connecting_manager;
+            connecting_manager = listening_manager;
+            listening_manager = temp_manager;
+            assert!(connecting_manager.add_peer(listening_id.clone(), addresses.clone()));
```

Comment:
\+ `assert!(!listening_manager.add_peer(connecting_id.clone(), addresses.clone()))`? :)

Reply:
We already did that in the `if` earlier, I don't think we have to check that again, it would be unnecessarily confusing.